charts/testkube-logs-service/templates/ingress-grpc.yaml

```diff
@@ -0,0 +1,64 @@
+{{- if and .Values.global.ingress.enabled .Values.grpcIngress.enabled -}}
+{{- $fullName := include "testkube-log-service.fullname" . -}}
+{{- if and .Values.ingress.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
+  {{- if not (hasKey .Values.grpcIngress.annotations "kubernetes.io/ingress.class") }}
+  {{- $_ := set .Values.grpcIngress.annotations "kubernetes.io/ingress.class" .Values.ingress.className}}
+  {{- end }}
+{{- end }}
+{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1
+{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: networking.k8s.io/v1beta1
+{{- else -}}
+apiVersion: extensions/v1beta1
+{{- end }}
+kind: Ingress
+metadata:
+  name: {{ $fullName }}-grpc
+  labels:
+    {{- include "testkube-log-service.labels" . | nindent 4 }}
+    {{- with .Values.grpcIngress.labels }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+  annotations:
+    nginx.ingress.kubernetes.io/backend-protocol: {{ if .Values.api.tls.serveHTTPS }}GRPCS{{ else }}GRPC{{ end }}
+    nginx.ingress.kubernetes.io/proxy-body-size: {{ .Values.grpcIngress.maxPayloadSize }}
+    nginx.ingress.kubernetes.io/client-header-timeout: "10800"
+    nginx.ingress.kubernetes.io/client-body-timeout: "10800"
+    {{- if and (not .Values.api.tls.serveHTTPS) (eq .Values.global.certificateProvider "cert-manager") }}
+    cert-manager.io/cluster-issuer: {{ required ".Values.global.certManager.issuerRef must be provided if provider is cert-manager" .Values.global.certManager.issuerRef }}
+    {{- end }}
+    {{- with .Values.ingress.annotations }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+    {{- with .Values.grpcIngress.annotations }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  {{- if and .Values.ingress.className (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion) }}
+  ingressClassName: {{ .Values.ingress.className }}
+  {{- end }}
+  tls:
+    - hosts:
+        - {{ include "testkube-log-service.ingress.grpcHost" . | quote }}
```

Comment:
The Ingress will point to `agent....`. is that right?

Reply:
yes I'm adding this one on path route of the agent - to no introduce another level of complexity. 

Ideally we should have API Gateway here to just add it - but here ingress will work for me like gateway.